choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerDBService.java

```diff
@@ -164,6 +195,65 @@ public ChoreographerStep addNextStepsToStep(final String stepName, final Set<Str
         }
     }
 
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerStepDetail addStepDetailToStep(ChoreographerStep step, String type, ChoreographerOFRRequestDTO ofrRequestDTO) {
```

Comment:
should be private
should be called from createStep, not from createAction
type should enum
ofrRequestDTO => orchestrationForm